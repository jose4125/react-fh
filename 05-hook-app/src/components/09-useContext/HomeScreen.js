import { useContext } from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {
  const { user } = useContext(UserContext);
  return (
    <div>
      <h1>home</h1>
      <pre>{JSON.stringify(user, null, 2)}</pre>
    </div>
  );
};
