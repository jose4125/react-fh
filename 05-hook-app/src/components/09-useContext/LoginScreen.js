import { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);
  return (
    <div>
      <h1>login</h1>
      <hr />
      <button onClick={() => setUser({ id: 2, name: "jose" })}>Login</button>
    </div>
  );
};
