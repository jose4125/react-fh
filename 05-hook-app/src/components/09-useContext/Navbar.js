import { NavLink } from "react-router-dom";

export const NavBar = () => {
  return (
    <nav>
      <ul>
        <li>
          <NavLink exact activeClassName="active" to="/">
            Home
          </NavLink>
        </li>
        <li>
          <NavLink activeClassName="active" to="/about">
            About
          </NavLink>
        </li>
        <li>
          <NavLink activeClassName="active" to="/login">
            Login
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};
