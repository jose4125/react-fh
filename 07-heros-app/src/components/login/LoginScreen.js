import { useHistory } from "react-router-dom";

const LoginScreen = () => {
  let history = useHistory();
  const handleLogin = () => {
    console.log("login");
    history.replce("/");
  };

  return (
    <div className="container mt-5">
      <h1>Login</h1>
      <hr />
      <button className="btn btn-primary" onClick={handleLogin}>
        Login
      </button>
    </div>
  );
};

export default LoginScreen;
