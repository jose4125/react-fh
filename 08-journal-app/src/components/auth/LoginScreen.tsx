import { RootState } from '../utils/types';
import Link from '../link/Link';
import Button from '../button/Button';
import { useForm } from '../hooks/useForm';
import { FormEvent } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startLoginEmailPassword, startLogingWithGoogle } from './authActions';

const LoginScreen = (): JSX.Element => {
  const dispatch = useDispatch();
  const { loading } = useSelector(({ ui }: RootState) => ui);
  const { formState, handleInputChange } = useForm({
    email: '',
    password: '',
  });

  const { email, password } = formState;

  const handleLogin = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    dispatch(startLoginEmailPassword(email, password));
  };

  const handleLoginWithGoogle = () => {
    dispatch(startLogingWithGoogle());
  };

  return (
    <div className="animate__animated animate__fadeIn animate__faster">
      <h3 className="auth__title">Login</h3>
      <form onSubmit={handleLogin} noValidate data-test="login-form">
        <input
          className="auth__input"
          type="email"
          name="email"
          placeholder="Email"
          autoComplete="off"
          value={email}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="password"
          name="password"
          placeholder="Password"
          autoComplete="off"
          value={password}
          onChange={handleInputChange}
        />
        <Button
          className="btn btn-primary btn-block"
          type="submit"
          disabled={loading}
        >
          Login
        </Button>
      </form>
      <hr />
      <div className="auth__social-networks">
        <p className="mb-1">Login with social network</p>
        <Button
          className="google-btn"
          onClick={handleLoginWithGoogle}
          data-test="login-with-google"
        >
          <div className="google-icon-wrapper">
            <img
              className="google-icon"
              src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg"
              alt="google button"
            />
          </div>
          <p className="btn-text">
            <b>Sign in with google</b>
          </p>
        </Button>
      </div>
      <Link className="link" to="/auth/register">
        Create new account
      </Link>
    </div>
  );
};

export default LoginScreen;
