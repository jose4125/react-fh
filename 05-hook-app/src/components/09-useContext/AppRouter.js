import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./Navbar";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <Switch>
          <Route exact path="/" component={HomeScreen} />
          <Route path="/about" component={AboutScreen} />
          <Route path="/login" component={LoginScreen} />
          <Redirect to="/" />
        </Switch>
      </div>
    </Router>
  );
};
