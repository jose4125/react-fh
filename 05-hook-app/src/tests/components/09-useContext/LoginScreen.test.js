import { mount } from "enzyme";
import { LoginScreen } from "../../../components/09-useContext/LoginScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe("test LoginScreen", () => {
  const setUser = jest.fn();
  const wrapper = mount(
    <UserContext.Provider
      value={{
        setUser,
      }}
    >
      <LoginScreen />
    </UserContext.Provider>
  );

  test("Should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should click the button and set the user", () => {
    wrapper.find("button").simulate("click");
    // const setUserFn = wrapper.find("button").prop("onClick");
    // setUserFn();
    expect(setUser).toHaveBeenCalled();
    expect(setUser).toHaveBeenCalledTimes(1);
    expect(setUser).toHaveBeenCalledWith({ id: 2, name: "jose" });
  });
});
