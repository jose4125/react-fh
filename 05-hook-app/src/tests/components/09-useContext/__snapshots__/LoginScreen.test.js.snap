// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LoginScreen Should render properly 1`] = `
<div>
  <h1>
    login
  </h1>
  <hr />
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
